### DISCUSSION TITLE
可以用bpf监听复制粘贴吗？

### DISCUSSION THREAD

**Discussion by xiaonuoz** [linuxdeepin/developer-center#4474] [Development Questions | 开发答疑] [answered]
类似监听文件读写那样。选中一段文字复制/粘贴后是否会调用某个内核方法，参数包含了文件内容和它的操作类型。请问有这个内核函数么？

**Comment by Decodetalkers** [comment]
那是剪切板吧。。x11是有现成的，gtk和qt都有包装，gtk是有，qt我不知道

wayland的话我没有研究

**Reply by xiaonuoz** [reply]
剪切板我知道的，可以用x11获取剪切的内容，但是我想的是从我这边开发的话，通过监听剪切板内容是否发生改变来判断是否复制的话我就得整个for死循环一直监听，我希望的是用户复制操作时，调用了某个内核函数，然后它主动来通知我发生了复制操作，不知道有这种内核函数么？

**Reply by xiaonuoz** [reply]
意思就是用户复制某段话时，有没有调用某个内核函数，这个内核函数包含了剪切的内容类型等，我直接用bpf去监听这个函数就行

**Reply by Decodetalkers** [reply, marked as answer]
> 剪切板我知道的，可以用x11获取剪切的内容，但是我想的是从我这边开发的话，通过监听剪切板内容是否发生改变来判断是否复制的话我就得整个for死循环一直监听，我希望的是用户复制操作时，调用了某个内核函数，然后它主动来通知我发生了复制操作，不知道有这种内核函数么？

https://doc.qt.io/qt-6/qclipboard.html#changed
